java/bulk-export/bulk-export-task-execution/src/test/java/sleeper/bulkexport/taskexecution/ECSBulkExportTaskRunnerLocalStackIT.java

```diff
@@ -93,35 +98,137 @@ void setUp() {
+        assertThat(getMessagesFromDlq().size()).isEqualTo(0);
+        assertThat(getMessagesFromQueue().size()).isEqualTo(1);
+    }
+
+    @Test
+    public void shouldReturnMessageToQueueAfterFailure() throws Exception {
+        // Given
+        String messageString = "This will cause a failure!";
+        configureJobQueuesWithMaxReceiveCount(2);
+        send(messageString);
+
+        // When
+        assertThatThrownBy(() -> runTask())
+                .hasMessageContaining("Expected BEGIN_OBJECT but was STRING");
+
+        // Then
+        assertThat(getMessagesFromDlq().size()).isEqualTo(0);
+        List<String> messages = getMessagesFromQueue();
+        assertThat(messages)
+                .size().isEqualTo(1);
+        assertThat(messages.get(0)).contains(messageString);
+    }
+
+    @Test
+    public void shouldHandleExceptionInProcessingAndSendToDlq() throws Exception {
+        // Given
+        configureJobQueuesWithMaxReceiveCount(1);
+        Record record1 = new Record(Map.of("key", 5, "value1", "5", "value2", "some value"));
+        Record record2 = new Record(Map.of("key", 15, "value1", "15", "value2", "other value"));
+        FileReference file = addPartitionFile("L", "file", List.of(record1, record2));
+        BulkExportLeafPartitionQuery query = createQueryWithIdsFilesAndBrokenPartition("e-id", "se-id", file);
+
         send(query);
 
         // When
+        assertThatThrownBy(() -> runTask())
+                .hasMessageContaining("Partition not found: NO-ID");
+
         runTask();
 
-        // Then the query region is ignored for now
-        assertThat(readOutputFile(query)).containsExactly(record1, record2);
-        assertThat(getMessagesFromQueue()).isEmpty();
+        // Then
+        List<String> messages = getMessagesFromDlq();
+        assertThat(messages)
+                .size().isEqualTo(1);
+
+        BulkExportLeafPartitionQuery receivedQuery = serDe.fromJson(messages.get(0));
+        assertThat(receivedQuery).isEqualTo(query);
+        assertThat(getMessagesFromQueue().size()).isEqualTo(0);
+
+    }
+
+    @Test
+    public void shouldMoveMessageToDlqAfterTwoFailures() throws Exception {
+        // Given
+        String messageString = "This will cause a failure!";
+        configureJobQueuesWithMaxReceiveCount(1);
+        send(messageString);
+
+        // When
+        // The task needs to be run twice for it to be moved to the DLQ
+        assertThatThrownBy(() -> runTask())
+                .hasMessageContaining("Expected BEGIN_OBJECT but was STRING");
+
+        runTask();
+
+        // Then
+        List<String> messages = getMessagesFromDlq();
+        assertThat(messages)
+                .size().isEqualTo(1);
+        assertThat(messages.get(0)).contains(messageString);
+        assertThat(getMessagesFromQueue().size()).isEqualTo(0);
     }
 
     @Test
     public void shouldProcessNoMessages() throws Exception {
```

Comment:
This test takes 20 seconds to run. We can reduce the wait time for tests. Before this PR it was set to 0.